packages/common/src/utils/arrays/utils.ts

```diff
@@ -1,15 +1,13 @@
-export const ifTheSameElseDefault = <
-  T,
-  K extends keyof T,
-  J extends T[K] | undefined
->(
+export const ifTheSameElseDefault = <T, K extends keyof T, J>(
   someEntities: T[],
   key: K,
   defaultValue: J
-): J => {
-  const value = someEntities[0]?.[key] as J;
-  const allTheSame = someEntities.every(entity => entity[key] === value);
-  return allTheSame && value ? value : defaultValue;
+): J | T[K] => {
+  const value = someEntities[0]?.[key];
+  const allTheSame = someEntities.every(entity => {
+    return entity[key] === value;
+  });
+  return allTheSame && value != undefined ? value : defaultValue;
```

Comment:
I've changed this to allow the default value to differ from the value of the `entity[key]` (as to use `[multiple]` for number values)